src/Stratis.Features.SystemContracts.Tests/EndToEndTests.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Stratis.Bitcoin.Networks;
+using Stratis.Features.SystemContracts.Contracts;
+using Stratis.Patricia;
+using Stratis.SmartContracts.Core.State;
+using Xunit;
+
+namespace Stratis.Features.SystemContracts.Tests
+{
+    public class EndToEndTests
+    {
+        [Fact]
+        public void Complete_Execution_Success()
+        {
+            var network = new StraxMain();
+            var authContract = new AuthContract.Dispatcher(network.SystemContractContainer);
```

Comment:
Can we increase the similarity between embedded smart contracts and on-chain smart contracts by making the same dispatch mechanism available to both - even if in one case the framework may be invoking it? I.e. increase the amount of common code being executed.

Reply:
We could but that's a substantial change to smart contracts and outside the scope of this work.

Reply:
It has the same scope as this more substantial changes with a large dispatcher implementation as well as mandatory dispatching code being added to the contracts themselves.

Reply:
The changes in this PR have no impact on the way contracts are written for Cirrus.

Reply:
> The changes in this PR have no impact on the way contracts are written for Cirrus.

That may be the goal and I understand the motivation (reducing changes to Smart Contracts) but it does go against code re-usability to not invent a common mechanism. I suppose you can justify it and put that justification up for scrutiny.